js/components/dataset/resource/file-form.vue

```diff
@@ -102,11 +102,25 @@ export default {
     },
     computed: {
         upload_endpoint: function() {
-            let endpoint = API.datasets.operations.upload_dataset_resource;
+            let operations = API.datasets.operations;
+            let params = {};
+            let endpoint = 'upload_';
+            if (typeof this.dataset !== 'undefined') {
+                params = {dataset: this.dataset.id};
+            }
             if (this.community) {
-                endpoint = API.datasets.operations.upload_community_resource;
+                endpoint += 'community_';
+                params.community = this.resource.id;
+            } else {
+                endpoint += 'dataset_';
             }
-            return endpoint.urlify({dataset: this.dataset.id});
+            if (this.resource.id) {
+                endpoint += 'resource';
+                params.rid = this.resource.id;
+            } else {
+                endpoint += 'resources';
+            }
```

Comment:
There are 4 cases from 2 conditions, I tried several approaches and couldn't figure out a better one but I'm open to suggestions. The issue with the current state of the code is that you have to follow the logic to recompose the endpoint which is far from being explicit…